src/sinks/datadog/metrics/sink.rs

```diff
@@ -145,3 +154,270 @@ where
+                }
+                _ => unreachable!("current index must represent a counter"),
+            }
+        }
+
+        idx += 1;
+    }
+
+    metrics
+}
+
+#[cfg(test)]
+mod tests {
+    use chrono::{DateTime, Utc};
+    use proptest::prelude::*;
+    use vector_core::event::{Metric, MetricKind, MetricValue};
+
+    use super::collapse_counters_by_series_and_timestamp;
+
+    fn arb_collapsible_metrics() -> impl Strategy<Value = Vec<Metric>> {
+        let ts = Utc::now();
+
+        any::<Vec<(u16, MetricValue)>>().prop_map(move |values| {
+            values
+                .into_iter()
+                .map(|(id, value)| {
+                    let name = format!("{}-{}", value.as_name(), id);
+                    Metric::new(name, MetricKind::Incremental, value).with_timestamp(Some(ts))
+                })
+                .collect()
+        })
+    }
+
+    fn create_counter(name: &str, value: f64) -> Metric {
+        Metric::new(
+            name,
+            MetricKind::Incremental,
+            MetricValue::Counter { value },
+        )
+    }
+
+    fn create_gauge(name: &str, value: f64) -> Metric {
+        Metric::new(name, MetricKind::Incremental, MetricValue::Gauge { value })
+    }
+
+    #[test]
+    fn collapse_no_metrics() {
+        let input = Vec::new();
+        let expected = input.clone();
+        let actual = collapse_counters_by_series_and_timestamp(input);
+
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn collapse_single_metric() {
+        let input = vec![create_counter("basic", 42.0)];
+        let expected = input.clone();
+        let actual = collapse_counters_by_series_and_timestamp(input);
+
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn collapse_identical_metrics_gauge() {
+        let input = vec![create_gauge("basic", 42.0), create_gauge("basic", 42.0)];
+        let expected = input.clone();
+        let actual = collapse_counters_by_series_and_timestamp(input);
+
+        assert_eq!(expected, actual);
+
+        let input = vec![
+            create_gauge("basic", 42.0),
+            create_gauge("basic", 42.0),
+            create_gauge("basic", 42.0),
+            create_gauge("basic", 42.0),
+            create_gauge("basic", 42.0),
+            create_gauge("basic", 42.0),
+            create_gauge("basic", 42.0),
+        ];
+        let expected = input.clone();
+        let actual = collapse_counters_by_series_and_timestamp(input);
+
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn collapse_identical_metrics_counter() {
+        let counter_value = 42.0;
+        let input = vec![
+            create_counter("basic", 42.0),
+            create_counter("basic", 42.0),
+            create_counter("basic", 42.0),
+            create_counter("basic", 42.0),
+            create_counter("basic", 42.0),
+            create_counter("basic", 42.0),
+            create_counter("basic", 42.0),
+        ];
```

Comment:
Should these `create_counter` just use the `counter_value` var?

Reply:
Oh, derp. Yes, they should.

Reply:
Fixed in a764ee8.